library/std/src/os/unix/net/stream.rs

```diff
@@ -633,7 +664,11 @@ impl io::Write for UnixStream {
 #[stable(feature = "unix_socket", since = "1.10.0")]
 impl<'a> io::Write for &'a UnixStream {
     fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
-        self.0.write(buf)
+        let len = cmp::min(buf.len(), <wrlen_t>::MAX as usize) as wrlen_t;
+        let ret = cvt(unsafe {
+            libc::send(self.0.as_raw(), buf.as_ptr() as *const c_void, len, MSG_NOSIGNAL)
+        })?;
+        Ok(ret as usize)
     }
```

Comment:
To allow for some possible future code reuse here, this could probably be a `send_with_flags` method on `Socket` to match https://github.com/rust-lang/rust/blob/4eca99a18eab3d4e28ed1ce3ee620d442955a470/library/std/src/sys/net/connection/socket/unix.rs#L284.

Reply:
yup. fixed.